lib/model/model.go

```diff
@@ -1738,32 +1738,33 @@ func sendIndexes(conn protocol.Connection, folder string, fs *db.FileSet, ignore
 // returns the highest sent sequence number.
 func sendIndexTo(prevSequence int64, conn protocol.Connection, folder string, fs *db.FileSet, ignores *ignore.Matcher, dbLocation string, dropSymlinks bool) (int64, error) {
 	deviceID := conn.ID()
-	batch := make([]protocol.FileInfo, 0, maxBatchSizeFiles)
-	batchSizeBytes := 0
 	initial := prevSequence == 0
 	var err error
 	var f protocol.FileInfo
+	batch := newFileInfoBatch(nil)
 	debugMsg := func(t string) string {
-		return fmt.Sprintf("Sending indexes for %s to %s at %s: %d files (<%d bytes) (%s)", folder, deviceID, conn, len(batch), batchSizeBytes, t)
+		return fmt.Sprintf("Sending indexes for %s to %s at %s: %d files (<%d bytes) (%s)", folder, deviceID, conn, len(batch.infos), batch.size, t)
 	}
-
-	fs.WithHaveSequence(prevSequence+1, func(fi db.FileIntf) bool {
-		if len(batch) == maxBatchSizeFiles || batchSizeBytes > maxBatchSizeBytes {
-			if initial {
-				if err = conn.Index(folder, batch); err != nil {
-					return false
-				}
-				l.Debugln(debugMsg("initial index"))
-				initial = false
-			} else {
-				if err = conn.IndexUpdate(folder, batch); err != nil {
-					return false
-				}
-				l.Debugln(debugMsg("batched update"))
+	msg := [2]string{"initial index", "batched update"}
+	batch.flushFn = func(fs []protocol.FileInfo) error {
+		if initial {
+			if err = conn.Index(folder, fs); err != nil {
+				return err
+			}
+			l.Debugln(debugMsg(msg[0]))
+			initial = false
+		} else {
+			if err = conn.IndexUpdate(folder, fs); err != nil {
+				return err
 			}
+			l.Debugln(debugMsg(msg[0]))
```

Comment:
`msg[1]`? Actually, just skip the debug logging, it serves not much purpose and isn't worth the complication of the `msg` stuff imho